docs/en/security-privacy-considerations.rst

```diff
@@ -0,0 +1,495 @@
+  where a list of mitigations from [`OAuthCrossDeviceSec`_] are discussed. Two examples are:
+  
+  - Short Lived/Timebound QR Codes: Reducing the lifetime of the QR code (e.g., 2-3 mins) is fundamental to restrict the time window available for the attacks. 
+  - One-Time QR Codes: One-Time QR codes restrict the possibility of attacks when the same QR code is sent to multiple victims. 
+
+SR-P-41
+~~~~~~~
+.. list-table:: 
+   :widths: 8 92
+
+   * - |check-icon|
+     - The protocol must ensure that an attacker cannot successfully forward an interaction between a Wallet and a Verifier to a Verifier under the control of the attacker.
+
+A prerequisite for a successful attack of this kind is that the attacker has access to some messages between the Wallet and the Verifier, for example, 
+the attacker might have access to the presentation contained in the VP Token. Given that, the proper implementation of TLS guarrantes confidentiality, avoiding the leakage of the response. 
+In addition to TLS, the existing implementation of **nonce** and audience checks in the presentation protocol should help to meet the P-41 security requirement. 
+Regarding the **nonce** claim, the Verifier MUST check that the **nonce** value in the VP Token matches the **nonce** value that is created by the Verifier during the authorization request.
+Regarding the **aud** value, the Verifier must check that the audience of the presentation matches the Verifier's identifier. 
+
+SR-P-50
+~~~~~~~
+.. list-table:: 
+   :widths: 8 92
+
+   * - |partially-check-iconcheck-icon| 
+     - The protocol must ensure that third parties cannot interfere with the binding process.
+
+In the issuance phase, the Holder binding happens at the Credential request to the protected Credential endpoint. This means that the attacker needs to obtain the access token 
+first and therefore send the request to the Credential endpoint and bind the Credentials to the keys under his control. The IT Wallet specification requires the use of a sender-constrained 
+access token, which means that the access token binds to the device using cryptographic materials. 
+
+The second surface for the attack is related to key management. In the case of using software-based keys, it is possible to clone the keys and move them to a device under 
+attacker control, and in the case of stealing the Credentials as well, the attacker can easily create proof of possession of the keys. Therefore, it is highly dependent on 
+the secure storage of the keys. In the case of the IT Wallet, it only supports the local internal WSCD that is providing Hardware bound keys that could mitigate the key cloning 
+attack, but it does not provide a high enough level of assurance to meet stringent eIDAS requirements (local internal WSCD has been shown to be insufficiently secure against 
+highly capable attackers). Therefore, the current implementation only partially satisfies it. 
```

Comment:
```suggestion
```
Please see https://github.com/italia/eudi-wallet-it-docs/pull/444/files#r1804753458 
This discussion must be brought to a new Issue

Reply:
For completeness  of the analysis and transparency we prefer not to delete the requirement. We have provided a new version with the required references and a clarification regarding the partially compliance of the requirement. Please check the new version.

Reply:
Hi @giadas we are totally ok with transparency, but we feel the current wording is misleading.

> the security of local internal WSCD solutions, such as TEE, is known to be vulnerable to attacks like buffer overflow attack and trustzone privilege escalation

That is technically correct, but in order to mount an attack against the TEE directly (abusing its interface with the secure monitor call) you need to be already at kernel level. As far as we know SMC isn't available from user-space, or in ARM parlance, SMC should be available only for "rings" EL1+.

It is true that one could attack the drivers instead (say /dev/qseecom on Android), but at least for the Android architecture afawk those devices are not reachable by generic, user installed applications as they are lacking the appropriate SELinux context. They are forced to go through the KeyStore Android daemon, adding another layer or protection from the OS.

In short, if you are at EL1+ or can subvert kernel level MAC at will (SELinux), there is no practical way to protect the secrets that will be given to Wallets on a WSCD Remote architecture. Given your assumption, the WSCD Remote architecture is equally broken too.

I've nothing to say on the certification side now that you have added a note clarifying that certification requirements are still being discussed at the EU Commission level.

Reply:
We agree that to mount an attack against the TEE you need to be at kernel level. However, as shown in the attack described in Section 4.1.1 of https://www.sciencedirect.com/science/article/pii/S0167404823000901#sec0009,  a user installed application (with zero permissions) is able to mount a privilege escalation attack by exploiting a vulnerable implementation of another application (for example the MediaServer Android application as shown in the paper).

We agree that if this attack happens, also a Remote WSCD could be compromised. However, here we are not assessing the security of the different WSCD types. We are mapping the requirement with the existing specification. Given the lack of certification profiles for local internal WSCD with high level of security, we suggest the partially compliance with this requirement. If the attack scenarios are misleading here, we can update the text in the following way:

> [...] The second surface for the attack is related to key management. In the case of using software-based keys, it is possible to clone the keys and move them to a device under attacker control, and in the case of stealing the credentials as well, the attacker can easily create proof of possession of the keys. IT Wallet is less vulnerable to these attacks as it supports local internal WSCD that uses hardware-based keys. However, the lack of a certification profile that certifies the local internal WSCD against highly capable attackers (the  certification for current TEE solutions on the market reaches AVA_VAN.3 at most [https://www.tuv-nederland.nl/assets/files/cerfiticaten/2021/08/nscib-cc-0244671-cr-1.0.pdf, https://globalplatform.org/specs-library/tee-protection-profile-v1-3/#]) makes the requirement only partially satisfied. 
> [...]

Is the new text ok?

Reply:
Yup, thanks for clarifying and the edit.